src/payment_call_builder.ts

```diff
@@ -36,12 +35,10 @@ export class PaymentCallBuilder extends CallBuilder<
    * @returns {PaymentCallBuilder} this PaymentCallBuilder instance
    */
   public forLedger(sequence: number | string): this {
-    this.filter.push([
+    return this.forEndpoint(
       "ledgers",
       typeof sequence === "number" ? sequence.toString() : sequence,
```

Comment:
(Not for this PR) FYI seems like the method `toString()` is available in strings too so we could simplify occurrences of this line to:

```suggestion
      sequence.toString(),
```

Reply:
I committed your suggestion but I'd like to do the rest in a separate PR to avoid scope creep as you said :+1: